migen/blink.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+"""Simple tri-colour LED blink example."""
+
+# Import lxbuildenv to integrate the deps/ directory
+import os, sys
+sys.path.insert(0, os.path.dirname(__file__))
+import lxbuildenv
+
+from migen import *
+import fomu
+
+if 'FOMU_REV' not in os.environ:
+    print('Board type must be specified in FOMU_REV environment variable!')
+    exit(1)
+
+#
+# Parameters from iCE40 UltraPlus LED Driver Usage Guide, pages 19-20
+#
+
+# Current modes
+RGBA_CURRENT_MODE_FULL = '0b0'
+RGBA_CURRENT_MODE_HALF = '0b1'
+
+# Current levels in Full / Half mode
+RGBA_CURRENT_04MA_02MA = '0b000001'
+RGBA_CURRENT_08MA_04MA = '0b000011'
+RGBA_CURRENT_12MA_06MA = '0b000111'
+RGBA_CURRENT_16MA_08MA = '0b001111'
+RGBA_CURRENT_20MA_10MA = '0b011111'
+RGBA_CURRENT_24MA_12MA = '0b111111'
+
+#
+# Signals
+#
+counter = Signal(28)
+
+rgb0_pwm = Signal(1)
+rgb1_pwm = Signal(1)
+rgb2_pwm = Signal(1)
+
+# Setup platform and correctly map pins for the
+# iCE40UP5K SB_RGBA_DRV hard macro.
```

Comment:
We should probably just use a dictionary or some other Python code here?

Reply:
```python
getattr(fomu, "Fomu{}Platform", os.environ.get('FOMU_REV', 'unknown').title())
```

Reply:
I'd rather leave it as-is, while elegant I think it might confuse people not too used to python?